### PR TITLE
fix: optimize intrusive types

### PR DESCRIPTION
**Pull request by romange**
Also, provide additional debugging variable inside connection to allow tracking of its thread assignment.

### CODE REVIEW

File: util/listener_interface.h [hunk 1]
```diff
@@ -61,10 +61,6 @@ class ListenerInterface {
   // Connection* are valid only during the call to cb.
   void TraverseConnections(TraverseCB cb);
 
-  // traverses all client connection in current thread. cb must adhere to rules from
-  // `TraverseConnections`;
-  void TraverseConnectionsOnThread(TraverseCB cb);
-
```
Comment (dranikpg) [inline util/listener_interface.h:67]: What's wrong with it? I need it to be public for Dragonfly 😢

File: util/fibers/detail/fiber_interface.h [hunk 2]
```diff
@@ -24,7 +24,7 @@ enum class Launch {
 namespace detail {
 
 using FI_ListHook =
-    boost::intrusive::slist_member_hook<boost::intrusive::link_mode<boost::intrusive::safe_link>>;
+    boost::intrusive::list_member_hook<boost::intrusive::link_mode<boost::intrusive::safe_link>>;
```
Comment (dranikpg) [inline util/fibers/detail/fiber_interface.h:27]: 👍🏻 I wonder how much it improves performance for heavy load with frequent disconnects 🙂

**Review by dranikpg — APPROVED** [review]

**Comment by romange** [comment]
Ah, I did not know that it's by design

**Comment by romange** [comment]
Can you please fix it, actually?

**Pull request merged**